### PR TITLE
Add physical exercise dataset with multiple targets

### PR DESCRIPTION
**Pull request by relf**
Dataset references:
* [scikit-learn](https://scikit-learn.org/stable/datasets/toy_dataset.html#linnerrud-dataset)
* [Tenenhaus, M. (1998). La regression PLS: theorie et pratique. Paris: Editions Technic.](https://books.google.fr/books?hl=fr&lr=&id=OesjK2KZhsAC&oi=fnd&pg=PA43&dq=Tenenhaus,+M.+(1998).+La+regression+PLS:+theorie+et+pratique.+Paris:+Editions+Technic.&ots=ExNsw80zCK&sig=BQkSlPc36v9GMy6oqhq0cHd5O9o#v=onepage&q=Linnerud&f=false)

Handling enabled in linfa with #88

### CODE REVIEW

File: datasets/src/lib.rs [hunk 1]
```diff
@@ -121,6 +127,33 @@ pub fn winequality() -> Dataset<f64, usize> {
         .with_feature_names(feature_names)
 }
 
+#[cfg(feature = "linnerud")]
+/// Read in the physical exercise dataset from dataset path.
+///
+/// Linnerud dataset contains 20 samples collected from 20 middle-aged men in a fitness club.
```
Comment (bytesnake) [inline datasets/src/lib.rs:133]: this is the smallest dataset I ever encountered :laughing:

**Review by bytesnake — COMMENTED** [review]
looking good to me, but according to Github your archive is empty. Can you re-order the struct in `src/iter.rs` so that clippy is happy, I don't know why this wasn't caught before

File: datasets/src/lib.rs (hunk 1 above)
Comment (relf) [reply datasets/src/lib.rs:133]: > but according to Github your archive is empty

I made the gz archives on Windows :grin:. I am going to use a Unix box...
> Can you re-order the struct in src/iter.rs

Ok

**Comment by relf** [comment]
I have integrated the linnerud dataset without thinking too much about its usage (till now! :laughing: ) and I have used a conversion of the integer targets as usize (shamelessly copying from wine quality stuff). 

But actually, I want to use the dataset as floats... Should we add some conversion helpers to `Dataset`? Or should I just remove the `as usize` conversion?

**Comment by bytesnake** [comment]
you probably don't want usize targets here, except when you want to do ordinal classification

**Review by bytesnake — APPROVED** [review]

**Pull request merged**